volatility3/framework/plugins/linux/mountinfo.py

```diff
@@ -0,0 +1,220 @@
+            requirements.ListRequirement(name="pids",
+                                         description="Filter on specific process IDs.",
+                                         element_type=int,
+                                         optional=True),
+            requirements.ListRequirement(name="mntns",
+                                         description="Filter results by mount namespace. "
+                                                     "Otherwise, all of them are shown.",
+                                         element_type=int,
+                                         optional=True),
+            requirements.BooleanRequirement(name="mount-format",
+                                            description="Shows a brief summary of the mount points information "
+                                            "with similar output format to the older /proc/[pid]/mounts or the "
+                                            "user-land command 'mount -l'.",
+                                            optional=True,
+                                            default=False),
+        ]
+
+    @classmethod
+    def _do_get_path(cls, mnt, fs_root) -> Union[None, str]:
+        """It mimics the Linux kernel prepend_path function."""
+        vfsmnt = mnt.mnt
+        dentry = vfsmnt.get_mnt_root()
+
+        path_reversed = []
+        while dentry != fs_root.dentry or vfsmnt.vol.offset != fs_root.mnt:
+            if dentry == vfsmnt.get_mnt_root() or dentry.is_root():
+                parent = mnt.get_mnt_parent().dereference()
+                # Escaped?
+                if dentry != vfsmnt.get_mnt_root():
+                    return None
+
+                # Global root?
+                if mnt.vol.offset != parent.vol.offset:
+                    dentry = mnt.get_mnt_mountpoint()
+                    mnt = parent
+                    vfsmnt = mnt.mnt
+                    continue
+
+                return None
+
+            parent = dentry.d_parent
+            dname = dentry.d_name.name_as_str()
+            path_reversed.append(dname.strip("/"))
+            dentry = parent
+
+        path = "/" + "/".join(reversed(path_reversed))
+        return path
+
+    @classmethod
+    def get_mountinfo(cls, mnt, task) -> Union[None, Tuple[int, int, str, str, str, List[str],
+                                                           List[str], str, str, List[str]]]:
+        """Extract various information about a mount point.
+        It mimics the Linux kernel show_mountinfo function.
+        """
+        mnt_root = mnt.get_mnt_root()
+        if not mnt_root:
+            return None
+
+        mnt_root_path = mnt_root.path()
+        superblock = mnt.get_mnt_sb()
+
+        mnt_id: int = mnt.mnt_id
+        parent_id: int = mnt.mnt_parent.mnt_id
+
+        st_dev = f"{superblock.major}:{superblock.minor}"
+
+        path_root = cls._do_get_path(mnt, task.fs.root)
+        if path_root is None:
+            return None
```

Comment:
would prefer this call to _do_get_path and the check of the return value to happen on line 91 so that the calls to path() and get_mnt_sb() can be avoided if the function is just going to bail out anyway. path() loops and can be slow

Reply:
done